tests/devices/unit_tests/test_shutter.py

```diff
@@ -0,0 +1,44 @@
+import pytest
+from ophyd_async.core import DeviceCollector, callback_on_mock_put, set_mock_value
+
+from dodal.devices.shutter import Shutter, ShutterSetState, ShutterState
+
+
+@pytest.fixture
+async def sim_shutter():
+    async with DeviceCollector(mock=True):
+        sim_shutter: Shutter = Shutter(
+            prefix="sim_shutter",
+            name="shutter",
+        )
+    return sim_shutter
+
+
+async def test_set_shutter_open(sim_shutter: Shutter):
+    desired_state: ShutterSetState = ShutterSetState.OPEN
+    callback_on_mock_put(
+        sim_shutter.position_set,
+        lambda _: set_mock_value(sim_shutter.position_readback, ShutterState.OPENING),
+    )
+    await sim_shutter.set(desired_state)
+    reading = await sim_shutter.read()
+
+    shutter_position = reading.get("shutter-position", {})
+    assert (
+        shutter_position["value"] is ShutterState.OPEN
+    ), f"Unexpected value: {shutter_position['value']}"
+
+
+async def test_set_shutter_close(sim_shutter: Shutter):
+    desired_state: ShutterSetState = ShutterSetState.CLOSE
+    callback_on_mock_put(
+        sim_shutter.position_set,
+        lambda _: set_mock_value(sim_shutter.position_readback, ShutterState.CLOSED),
```

Comment:
nit: I'd prefer to see this on the fixture, with sensible open/open close/closed

Reply:
ok but it's not passing the test. it might have to do with ophyd-async
![Uploading image.png…]()

Reply:
not sure how this is supposed to go @DiamondJoseph